features/netsocket/DNS.h

```diff
@@ -83,6 +101,33 @@ class DNS {
     virtual nsapi_value_or_error_t gethostbyname_async(const char *host, hostbyname_cb_t callback, nsapi_version_t version = NSAPI_UNSPEC,
                                                        const char *interface_name = NULL) = 0;
 
+    /** Translate a hostname to the multiple IP addresses (asynchronous)
+     *
+     *  The hostname may be either a domain name or an IP address. If the
+     *  hostname is an IP address, no network transactions will be performed.
+     *
+     *  If no stack-specific DNS resolution is provided, the hostname
+     *  will be resolved using a UDP socket on the stack.
+     *
+     *  The call is non-blocking. The result of the DNS operation is returned by the callback.
+     *  If this function returns failure, the callback will not be called. If it is successful,
+     *  (the IP address was found from the DNS cache), the callback will be called
+     *  before the function returns.
+     *
+     *  @param host     Hostname to resolve.
+     *  @addr_count     number of  expected IP addresses.
+     *  @param callback Callback that is called to return the result.
+     *  @param version  IP version of address to resolve. NSAPI_UNSPEC indicates that the
+     *                  version is chosen by the stack (defaults to NSAPI_UNSPEC).
+     *  @param interface_name  Network interface name
+     *  @return         NSAPI_ERROR_OK on immediate success,
+     *                  negative error code on immediate failure or
+     *                  a positive unique ID that represents the hostname translation operation
+     *                  and can be passed to cancel.
+     */
+    virtual nsapi_value_or_error_t getaddrinfo_async(const char *host, hostbyname_cb_t callback, nsapi_version_t version = NSAPI_UNSPEC,
```

Comment:
Why this is called `getaddrinfo_*` but it does not use the `SocketAddress *hints` like the non async version?

The whole point of `getaddrinfo` is that you give it some hints, and receive a results. We could receive results asynchronously, that is fine, but how to give the hints should be the same for both API.

Reply:
Added hints to getaddrinfo_async.